src/javascript/request-composer/components/request-composer.js

```diff
@@ -0,0 +1,614 @@
+    gapi.client.analyticsreporting.reports.batchGet(request
+      ).then(function(response) {
+        actions.updateResponse(response);
+        actions.updateSettings({
+          'responseType': settings.requestType,
+          'responseCohortSize': params.cohortSize
+      });
+      }, function(response) {
+        AlertDispatcher.addOnce({
+          title: 'Oops, there was an error',
+          message: response.result.error.message
+        });
+        actions.updateResponse(response);
+    });
+
+    actions.setQueryState(false);
+  }
+
+
+  /**
+   * React lifecycyle method below:
+   * http://facebook.github.io/react/docs/component-specs.html
+   * ---------------------------------------------------------
+   */
+  render() {
+
+    let {
+      isQuerying,
+      params,
+      response,
+      settings,
+      select2Options
+    } = this.props;
+
+    let formControlClass = 'FormControl FormControl--inline';
+    let formActionClass = formControlClass + ' FormControl--action';
+    let requiredFormControlClass = formControlClass +' FormControl--required';
+    let maximumSelectionSize = 2;
+
+    return (
+      <div>
+        <h3 className="H3--underline">Select a view</h3>
+
+        <ViewSelector
+          viewId={params.viewId}
+          onChange={this.handleViewSelectorChange} />
+
+        <Tabs
+          onSelect={this.handleRequestChange}
+          selectedIndex={REQUEST_TYPES.indexOf(settings.requestType)}
+        >
+          <TabList>
+            <Tab>Histogram Request</Tab>
+            <Tab>Pivot Request</Tab>
+            <Tab>Cohort Request</Tab>
+          </TabList>
+          <TabPanel>
+            <p>For dimensions with integer values, it is easier to understand
+              their characteristics by bucketing their values into ranges.
+            </p>
+          </TabPanel>
+          <TabPanel>
+            <p>Google Analytics Reporting API V4 allows you to generate pivot
+              tables. To construct a request with a pivot table, define the
+              pivot field in the ReportRequest.
+            </p>
+          </TabPanel>
+          <TabPanel>
+            <p>A cohort is a group of users who share a common characteristic.
+              For example, all users with the same Acquisition Date belong to
+              the same cohort. The cohort analysis report lets you isolate and
+              analyze cohort behavior.
+            </p>
+          </TabPanel>
+        </Tabs>
+
+        <h3 className="H3--underline">Set the query parameters</h3>
+
+        <form onSubmit={this.handleSubmit}>
+
+          <div className={requiredFormControlClass}>
+            <label className="FormControl-label">viewId</label>
+            <div className="FormControl-body">
+              <div className="FlexLine">
+                <input
+                  className="FormField FormFieldCombo-field"
+                  name="viewId"
+                  value={params.viewId}
```

Comment:
I'm getting the following error that's due to these `value` props not being defined:

> Warning: RequestComposer is changing an uncontrolled input of type undefined to be controlled. Input elements should not switch from uncontrolled to controlled (or vice versa). Decide between using a controlled or uncontrolled input element for the lifetime of the component. More info: https://fb.me/react-controlled-components

In the Query Explorer I use `someValue || ''` everywhere. You could do that or use the `defaultValue` prop. I'm not sure which is more idiomatic.